src/components/TagSet/TagSet.test.tsx

```diff
@@ -270,23 +279,27 @@ describe("TagSet Filter", () => {
   });
 
   it("returns the clicked tag's label to the `onClick` function", () => {
-    let currentLabel = "";
-    const onClick = (selectedLabel: string) => (currentLabel = selectedLabel);
+    let currentTag = {};
+    const onClick = (selectedTag) => (currentTag = selectedTag);
     render(
       <TagSet
         isDismissible
         onClick={onClick}
-        tagSetData={filterTagSetData.simple}
+        tagSetData={filterWithMoreProperties}
         type="filter"
       />
     );
 
     screen.getByText("Blue").click();
-    expect(currentLabel).toEqual("Blue");
+    expect(currentTag).toEqual({ label: "Blue", id: "blue", field: "color" });
```

Comment:
great! i'm surprised the typing on the onClick doesn't complain about the extra properties.

Reply:
I added `[key: string]: string;` to the prop definition so it's somewhat open for more values (it just won't work if it was `field: 2` since that's a number...)